jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/SsoSessionConfigurationManager.kt

```diff
@@ -0,0 +1,48 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.credentials
+
+import com.intellij.openapi.util.io.FileUtil
+import software.amazon.awssdk.profiles.ProfileFileLocation
+import software.amazon.awssdk.profiles.ProfileProperty.SSO_ACCOUNT_ID
+import software.amazon.awssdk.profiles.ProfileProperty.SSO_REGION
+import software.amazon.awssdk.profiles.ProfileProperty.SSO_ROLE_NAME
+import software.amazon.awssdk.profiles.ProfileProperty.SSO_START_URL
+import software.aws.toolkits.core.credentials.extractOrgID
+import software.aws.toolkits.jetbrains.core.credentials.SsoProfileConstants.SSO_SESSION_PROFILE_NAME
+import software.aws.toolkits.jetbrains.core.credentials.profiles.SsoSessionConstants.PROFILE_SSO_SESSION_PROPERTY
+import software.aws.toolkits.jetbrains.core.credentials.profiles.SsoSessionConstants.SSO_REGISTRATION_SCOPES
+import software.aws.toolkits.jetbrains.core.credentials.profiles.SsoSessionConstants.SSO_SESSION_SECTION_NAME
+
+class SsoSessionConfigurationManager {
+
+    val profileFile = ProfileFileLocation.configurationFilePath().toFile()
+
+    // writing sso-session format to the config file
+    fun writeSsoSessionProfileToConfigFile(ssoRegion: String, startUrl: String, scopes: List<String>, accountId: String, roleName: String) {
```

Comment:
what if a `sso-session` block of the same time already exists and we just want to update it?

Reply:
Sure, will fix this case